inspire_classifier/api.py

```diff
@@ -0,0 +1,180 @@
+    assert os.path.exists(os.path.join(current_app.config['CLASSIFIER_LANGUAGE_MODEL_DATA_DIR'], 'lbl_trn.npy'))
+    assert os.path.exists(os.path.join(current_app.config['CLASSIFIER_LANGUAGE_MODEL_DATA_DIR'], 'lbl_val.npy'))
+
+    map_and_save_tokens_to_ids_for_language_model(
+        current_app.config['CLASSIFIER_LANGUAGE_MODEL_DATA_DIR'], current_app.config['CLASSIFIER_DATA_ITOS_PATH'],
+        current_app.config['CLASSIFIER_MAXIMUM_VOCABULARY_SIZE'], current_app.config['CLASSIFIER_MINIMUM_WORD_FREQUENCY']
+    )
+    assert os.path.exists(current_app.config['CLASSIFIER_DATA_ITOS_PATH'])
+    assert os.path.exists(os.path.join(current_app.config['CLASSIFIER_LANGUAGE_MODEL_DATA_DIR'], 'trn_ids.npy'))
+    assert os.path.exists(os.path.join(current_app.config['CLASSIFIER_LANGUAGE_MODEL_DATA_DIR'], 'val_ids.npy'))
+
+    generate_and_save_classifier_tokens(current_app.config['CLASSIFIER_CLASSIFIER_DATA_DIR'])
+    assert os.path.exists(os.path.join(current_app.config['CLASSIFIER_CLASSIFIER_DATA_DIR'], 'tok_trn.npy'))
+    assert os.path.exists(os.path.join(current_app.config['CLASSIFIER_CLASSIFIER_DATA_DIR'], 'tok_val.npy'))
+    assert os.path.exists(os.path.join(current_app.config['CLASSIFIER_CLASSIFIER_DATA_DIR'], 'lbl_trn.npy'))
+    assert os.path.exists(os.path.join(current_app.config['CLASSIFIER_CLASSIFIER_DATA_DIR'], 'lbl_val.npy'))
+
+    map_and_save_tokens_to_ids_for_classifier(
+        current_app.config['CLASSIFIER_CLASSIFIER_DATA_DIR'], app.config['CLASSIFIER_DATA_ITOS_PATH']
+    )
+    assert os.path.exists(os.path.join(current_app.config['CLASSIFIER_CLASSIFIER_DATA_DIR'], 'trn_ids.npy'))
+    assert os.path.exists(os.path.join(current_app.config['CLASSIFIER_CLASSIFIER_DATA_DIR'], 'val_ids.npy'))
+
+
+def finetune_and_save_language_model():
+    '''
+    Finetunes the pretrained (on wikitext103) language model on our dataset.
+    '''
+    assert os.path.exists(current_app.config['CLASSIFIER_DATA_ITOS_PATH'])
+    assert os.path.exists(os.path.join(current_app.config['CLASSIFIER_LANGUAGE_MODEL_DATA_DIR'], 'trn_ids.npy'))
+    assert os.path.exists(os.path.join(current_app.config['CLASSIFIER_LANGUAGE_MODEL_DATA_DIR'], 'val_ids.npy'))
+    language_model = LanguageModel(
+        os.path.join(current_app.config['CLASSIFIER_LANGUAGE_MODEL_DATA_DIR'], 'trn_ids.npy'),
+        os.path.join(current_app.config['CLASSIFIER_LANGUAGE_MODEL_DATA_DIR'], 'val_ids.npy'),
+        current_app.config['CLASSIFIER_LANGUAGE_MODEL_DATA_DIR'], current_app.config['CLASSIFIER_DATA_ITOS_PATH']
+    )
+
+    if not os.path.exists(current_app.config['CLASSIFIER_PRETRAINED_LANGUAGE_MODEL_PATH']):
+        wikitext103_language_model_request = requests.get(current_app.config['CLASSIFIER_WIKITEXT103_LANGUAGE_MODEL_URL'],
+                                                       allow_redirects = True)
+        open(current_app.config['CLASSIFIER_PRETRAINED_LANGUAGE_MODEL_PATH'], 'wb').write(
```

Comment:
* minor: the result of a `request.get` is a response, not a request, so the naming is wrong.
* you should call `wikitext103_language_model_request.raise_for_status()` to make sure you're not writing an error page.
* you should never write bare `open`, but use it as a context manager.